### PR TITLE
Run sonobuoy plugin on AWS

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/19085

This removes the `apps` pipeline and `task`, while adding a new `aws-e2e` pipeline that would run the same tests that the `azure` pipeline runs, but on aws.

### CODE REVIEW

File: tekton/pipelines/apps.yaml [hunk 1]
```diff
@@ -1,93 +0,0 @@
-apiVersion: tekton.dev/v1beta1
```
Comment (MarcelMue) [inline tekton/pipelines/apps.yaml:1]: Why is all this getting deleted?

Comment (fiunchinho) [reply tekton/pipelines/apps.yaml:1]: It's explained in the issue linked above.

We have a sonobuoy plugin that runs a set of go tests. At some point a new test was added to test also the app platform. A new pipeline was added to run the sonobuoy plugin filtering out all the tests but the new app platform test. Instead, we should run all tests possible on all providers. I don't think there is a reason to run only the app platform test when we could be running the app platform test, the autoscaler test, connectivity test, etc.

File: tekton/pipelines/capz.yaml [hunk 2]
```diff
@@ -55,8 +55,10 @@ spec:
   - name: test-azure
     runAfter: [wait-for-condition-ready]
     taskRef:
-      name: azure
+      name: e2e
```
Comment (whites11) [inline tekton/pipelines/capz.yaml:58]: this is still azure, isn't it?

File: tekton/pipelines/debug-azure.yaml [hunk 3]
```diff
@@ -44,8 +44,10 @@ spec:
   - name: test-azure
     runAfter: [wait-for-ready]
     taskRef:
-      name: azure
```
Comment (whites11) [inline tekton/pipelines/debug-azure.yaml:47]: this is still azure, isn't it

**Review by whites11 — APPROVED** [review]
LGTMIIWFY

File: tekton/pipelines/capz.yaml (hunk 2 above)
Comment (fiunchinho) [reply tekton/pipelines/capz.yaml:58]: I only changed the task name.

**Review by njuettner — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]

**Review by MarcelMue — APPROVED** [review]
I understand now

**Pull request merged**